src/copas.lua

```diff
@@ -262,19 +360,35 @@ function copas.receive(client, pattern, part)
   local s, err
   pattern = pattern or "*l"
   local current_log = _reading_log
+  sto_timeout(client, "read")
+
   repeat
     s, err, part = client:receive(pattern, part)
-    if s or (not _isTimeout[err]) then
+
+    if s then
+      current_log[client] = nil
+      sto_timeout()
+      return s, err, part
+
+    elseif sto_timed_out() then
+      current_log[client] = nil
+      return nil, "timeout"
```

Comment:
I thought about that one, but actually it is the LuaSocket implementation that is wrong here. Still have hopes that LuaSocket will be fixed.

Reply:
actually the error string is simply there, so it can simply be returned. Only side effect is that in sporadic cases it might return the openssl errors "wantread" or "wantwrite" as well.

Reply:
revisiting this. Just updated the docs; https://github.com/keplerproject/copas/pull/97/commits/66c1f595d6e57d2991893cbcee76d452db68bd72

That's ugly, it is putting the burden on the user. Especially since `connect` also has corner cases. I see the benefit of easier porting, but this would mean that each Copas application would have to deal with all those differences on its own.

So I tend to actually go the opposite and just return "timeout" everywhere. What do you think @azdle?

Reply:
Maybe it can be made configurable, on a per co-routine basis?

Reply:
added a commit to make it configurable: https://github.com/keplerproject/copas/pull/97/commits/8127dd882ff5be5877c72b2566ce2b70653dddbb

feedback appreciated

Reply:
Making it configurable seems like a great option. :+1: